contracts/RelayHub.sol

```diff
@@ -251,18 +252,23 @@ contract RelayHub is IRelayHub {
 
         require(msg.sender == address(this), "Only RelayHub should call this function");
 
+        bool successPrePost;
+        bytes memory preRetVal;
+        bytes memory transaction = abi.encodeWithSelector(IRelayRecipient(to).preRelayedCall.selector, relayAddr, from, encodedFunction, (gasOverhead + initialGas - gasleft()), transactionFee);
+        (successPrePost,preRetVal) = to.call.gas(preRelayedCallMaxGas)(transaction);
+        require(successPrePost, "preRelayedCall reverted - Not calling the relayed transaction");
```

Comment:
A question - would it be possible to make this function return an integer, just as 'acceptRelayCall' does, so that in 'relayCall' we would know which function reverted - 'pre', 'relay' or 'post'? If it is, is there a reason not to do that now, that we are changing it? It will be a small bit of useful debug info on the world of "something went wrong. change something and try again"-based debugging :-)

Reply:
This is very tricky and adds a lot of complexity. The reason is that we have to revert all three methods (pre-, relayed call, post-) together so it's very hard to do that and still get a return value.